app/client/src/sagas/ActionExecution/PluginActionSaga/onPageLoadSaga.ts

```
@@ -0,0 +1,397 @@
+}
+
+function* executePageLoadAction(
+  pageAction: PageAction,
+  span?: Span,
+  actionExecutionContext?: ActionExecutionContext,
+) {
+  const currentEnvDetails: { id: string; name: string } = yield select(
+    getCurrentEnvironmentDetails,
+  );
+
+  if (pageAction.hasOwnProperty("collectionId")) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)